base_layer/core/src/covenants/arguments.rs

```diff
@@ -225,6 +240,8 @@ impl CovenantArg {
         }
     }
 
+    /// Require that the given `CovenantArg` instance is of the form
+    /// `CovenantArg::Uint` and extracts the underlying value.
```

Comment:
We can also replace `pub fn require_uint(self) -> Result<u64, CovenantError> {` with
```suggestion
require_x_impl!(require_uint, Uint, "uint", u64);
```
(but this was not due to the change in the macro, the original `$output_type:ident` handled it)